src/server/generic_family.cc

```diff
@@ -195,6 +251,36 @@ OpStatus OpPersist(const OpArgs& op_args, string_view key) {
   }
 }
 
+OpResult<std::string> OpDump(const OpArgs& op_args, string_view key) {
+  auto& db_slice = op_args.shard->db_slice();
+  auto [it, expire_it] = db_slice.FindExt(op_args.db_cntx, key);
+
+  if (IsValid(it)) {
+    DVLOG(1) << "Dump: key '" << key << "' successfully found, going to dump it";
+    std::unique_ptr sink = std::make_unique<BytesSink>();
+    RdbSerializer serializer(sink.get());
+
+    // According to Redis code we need to
+    // 1. Save the value itself - without the key
+    // 2. Save footer: this include the RDB version and the CRC value for the message
+    unsigned obj_type = it->second.ObjType();
+    unsigned encoding = it->second.Encoding();
+    auto type = RdbObjectType(obj_type, encoding);
+    DVLOG(1) << "We are going to dump object type: " << type;
+    std::error_code ec = serializer.WriteOpcode(type);
+    CHECK(!ec);
+    ec = serializer.SaveValue(it->second);
+    CHECK(!ec);  // make sure that fully was successful
+    ec = serializer.FlushMem();
+    CHECK(!ec);  // make sure that fully was successful
+    auto dump = AppendFooter(sink->value);
+    CHECK(dump.size() > 10);
```

Comment:
nit: `CHECK_GT(dump.size(), 10u);` is more informative

Reply:
done